src/controllers/item.js

```diff
@@ -4,7 +4,8 @@ module.exports = {
   getAllItems: async (req, res) => {
     try {
       const items = await ItemModel.find();
-      res.json(items);
+      if (items.length < 0) res.json({ message: 'No items found' });
+      else res.json(items);
```

Comment:
To make it easier for front-end developers to differentiate between when there are items and when there are none, throwing an error would be the best practice. Checking the response code would be enough. 
Another thing is that when items length is zero, it will go to the else statement, instead of less than it should be less than or equal to zero.

Reply:
Thank you @AmmarBaki2 for noting that out. All resolved.

Reply:
I have a different perspective about returning error when you have an empty array. It's totally fine to return an error as Ammar said, but I think it's better to return either null or the empty array itself because  it's not an error and the frontend developer can notice the difference between an error and empty array.
You can check this stackoverflow question

https://stackoverflow.com/questions/24852784/return-empty-array-or-error/24852870#24852870

Reply:
It agree with you Mahmoud that it is not an error. Thank you for sharing your perspective. 
Let's see Shrreya's opinion.. @Shrreya what front end developers would like to get for an empty array?

Reply:
@Peri7at @mkkasem @AmmarBaki2 Interesting discussion!

Firstly, I agree that an empty array is not an error. The server did what it was requested to, there's just no content to return to the client. So it should be a regular 200 OK response code, not an error response code.

But coming to @AmmarBaki2's point about using an error code that helps frontend devs to differentiate, this is usually an internal code, not an HTTP error response code. Or sometimes backend devs return the array length along with the array as extra information in the response body.

I think you can keep it as it for now, frontend devs can check the array length to handle no items scenario.

Reply:
Thank you for the info @Shrreya. So I will change back to _**if (items.length <= 0) res.json({ message: 'No items found' });**_, right? Right now it is _**if (items.length <= 0) throw new Error('No items found');**_

Reply:
@Peri7at No change required.
The error thrown will go to catch block and then res.json will be called there.